scanpy/datasets/_expression_atlas.py

```diff
@@ -0,0 +1,174 @@
+
+    experiment_dir = settings.dataset_dir / accession
+    experiment_dir.mkdir(parents=True, exist_ok=True)
+
+    with tqdm(
+        unit="B",
+        unit_scale=True,
+        unit_divisor=1024,
+        miniters=1,
+        desc="experimental_design.tsv",
+    ) as t:
+        urlretrieve(
+            base_url + sampledata_path,
+            experiment_dir / "experimental_design.tsv",
+            reporthook=tqdm_hook(t),
+        )
+    with tqdm(
+        unit="B",
+        unit_scale=True,
+        unit_divisor=1024,
+        miniters=1,
+        desc="expression_archive.zip",
+    ) as t:
+        urlretrieve(
+            base_url + quantification_path,
+            experiment_dir / "expression_archive.zip",
+            reporthook=tqdm_hook(t),
+        )
+
+
+def read_mtx_from_stream(stream):
+    stream.readline()
+    n, m, _ = (int(x) for x in stream.readline()[:-1].split(b" "))
+    data = pd.read_csv(
+        stream,
+        sep=r"\s+",
+        header=None,
+        dtype={0: np.integer, 1: np.integer, 2: np.float32},
+    )
+    mtx = sparse.csr_matrix((data[2], (data[1] - 1, data[0] - 1)), shape=(m, n))
+    return mtx
+
+
+def read_expression_from_archive(archive: ZipFile):
+    info = archive.infolist()
+    assert len(info) == 3
+    mtx_data_info = [i for i in info if i.filename.endswith(".mtx")][0]
+    mtx_rows_info = [i for i in info if i.filename.endswith(".mtx_rows")][0]
+    mtx_cols_info = [i for i in info if i.filename.endswith(".mtx_cols")][0]
+    with archive.open(mtx_data_info, "r") as f:
+        expr = read_mtx_from_stream(f)
+    with archive.open(mtx_rows_info, "r") as f:
+        varname = pd.read_csv(f, sep="\t", header=None)[1]
+    with archive.open(mtx_cols_info, "r") as f:
+        obsname = pd.read_csv(f, sep="\t", header=None)[1]
+    adata = anndata.AnnData(expr)
+    adata.var_names = varname
+    adata.obs_names = obsname
+    return adata
+
+
+def expression_atlas(accession: str, *, filter_boring: bool = False):
+    """
+    Load a dataset from the `EBI Single Cell Expression Atlas`_.
+
+    Params
+    ------
+    accession:
+        Dataset accession. Like ``E-GEOD-98816`` or ``E-MTAB-4888``.
+    filter_boring:
+        Whether boring labels in obs should be automatically removed.
+
+    Example
+    -------
+    >>> adata = sc.datasets.expression_atlas("E-MTAB-4888")
+
+    .. _`EBI Single Cell Expression Atlas`: https://www.ebi.ac.uk/gxa/sc/experiments
+    """
+    experiment_dir = settings.dataset_dir / accession
+    dataset_path = experiment_dir / "{}.h5ad".format(accession)
+    try:
+        adata = anndata.read(dataset_path)
+        if filter_boring:
+            adata.obs = _filter_boring(adata.obs)
+        return adata
+    except OSError:
+        # Dataset couldn't be read for whatever reason
+        pass
+
+    download_experiment(accession)
+
+    print("Downloaded {} to {}".format(accession, experiment_dir.absolute()))
```

Comment:
Hm, the default logging level is `warning`, not `info`.

It's strange to use `logg.warn` as we don't want to print `WARNING`, so, I guess, simply using print is fine... (it won't appear in a logging file, but no one is using this functionality anyway, so we can drop it...)

Can you add a comment on that one can change `settings.datasetsdir`?

Reply:
I'd gone with printing for downloading since I believe the progress bars go to stdout anyways, so it's not adding noise to a previously silent operation.

Also, I had been thinking about trying to make logging a bit more consistent and easy to implement (#472), and might pick that up once I've got bandwidth.

Should the comment about `settings.datasetdir` go into the function docs or the printed message?

Reply:
I thought about the printed message...

Making logging more consistent would be terrific. :D